src/status_im/chat/commands/impl/transactions.cljs

```diff
@@ -292,37 +297,57 @@
   protocol/Yielding
   (yield-control [_ {{{amount :amount asset :asset} :params} :content} {:keys [db] :as cofx}]
     ;; Prefill wallet and navigate there
-    (let [recipient-contact     (or
-                                 (get-in db [:contacts/contacts (:current-chat-id db)])
-                                 (db.contact/public-key->new-contact (:current-chat-id db)))
-
-          sender-account        (:account/account db)
-          chain                 (keyword (:chain db))
-          symbol-param          (keyword asset)
-          all-tokens            (:wallet/all-tokens db)
-          {:keys [symbol decimals]} (tokens/asset-for all-tokens chain symbol-param)
-          {:keys [value error]}     (wallet.db/parse-amount amount decimals)
-          next-view-id              (if (:wallet-set-up-passed? sender-account)
-                                      :wallet-send-modal-stack
-                                      :wallet-send-modal-stack-with-onboarding)]
-      (fx/merge cofx
-                {:db (-> db
-                         (update-in [:wallet :send-transaction]
-                                    assoc
-                                    :amount (money/formatted->internal value symbol decimals)
-                                    :amount-text amount
-                                    :amount-error error)
-                         (choose-recipient.events/fill-request-details
-                          (transaction-details recipient-contact symbol) false)
-                         (update-in [:wallet :send-transaction]
-                                    dissoc :id :password :wrong-password?))
-                 ;; TODO(janherich) - refactor wallet send events, updating gas price
-                 ;; is generic thing which shouldn't be defined in wallet.send, then
-                 ;; we can include the utility helper without running into circ-dep problem
-                 :update-gas-price {:web3          (:web3 db)
-                                    :success-event :wallet/update-gas-price-success
-                                    :edit?         false}}
-                (navigation/navigate-to-cofx next-view-id {}))))
+    (if (= asset "STT")
```

Comment:
could you elaborate on why we check "STT" string here, and is it possible to split this function, it's too large

Reply:
or leave a comment, is it temporary duplicate logic?

Reply:
yes it is temporary duplicated logic, right now the wallet sign transaction flow is garbage so this goes around it for STT transactions. It will need to be changed when we go on mainnet and the best time for that is once you implemented the new sign transaction screen which avoids all the wallet flow